src/spi.rs

```diff
@@ -779,6 +779,65 @@ macro_rules! hal {
                     Spi { spi, pins }
                 }
 
+                /// Enable transmit register empty interrupt
+                pub fn enable_txe_interrupt(&mut self) {
+                    self.spi.cr2.modify(|r, w| unsafe {w.bits(r.bits() | (1 << 7))});
```

Comment:
This should work, but a more canonical (and saner) version would be:

```suggestion
                    self.spi.cr2.modify(|_, w| { w.txeie().set_bit() });
```

You only need to use the `r` proxy part if you're actually interested in the value e.g. for decision making or because the SVD file is broken so you have to work with the raw value.

Reply:
Oh, that's much prettier. I didn't know this works.